test/functional/autocmd/tabnewentered_spec.lua

```diff
@@ -29,3 +37,351 @@ describe('TabNewEntered', function()
     end)
   end)
 end)
+
+describe('tabpage/earlier', function()
+  before_each(clear)
+  local function switches_to_earlier_after_new_tab_creation_at_end(characters)
+    return function()
+        -- Add three tabs for a total of four
+        command('tabnew')
+        command('tabnew')
+        command('tabnew')
+
+        -- The earlier tab is now the third.
+        eq(3, eval('tabpagenr(\'#\')'))
+
+        -- Switch to the earlier (third) tab
+        feed(characters)
+
+        eq(dedent([=[
+
+
+          Tab page 1
+              [No Name]
+          Tab page 2
+              [No Name]
+          Tab page 3
+          >   [No Name]
+          Tab page 4
+              [No Name]]=]),
+           redir_exec('tabs')
+        )
+
+        -- The earlier tab is now the fourth.
+        eq(4, eval('tabpagenr(\'#\')'))
+    end
+  end
+  it('switches to earlier via g<Tab> after new tab creation at end', switches_to_earlier_after_new_tab_creation_at_end('g<Tab>'))
```

Comment:
`switches_to_earlier_after_new_tab_creation_at_end` is longer than we name most functions, but if we're going to keep that name we would at least add a line break here:

```suggestion
  it('switches to earlier via g<Tab> after new tab creation at end',
    switches_to_earlier_after_new_tab_creation_at_end('g<Tab>'))
```

Reply:
Fixed.